### PR TITLE
Handle failure in from_index

### LINKED ISSUE #4: Handle MIDIGetDestination/MIDIGetSource failure
If `MIDIGetDestination` or `MIDIGetSource` fail (e.g. because an invalid port number was given), they return `NULL`, which is not handled in `from_index`. Instead, `from_index` should return a `Result`.

I'm going to prepare a PR if you don't beat me to it.

### PR DESCRIPTION
**Pull request by Boddlnagg**
Fixes #4. I decided to return an `Option` instead of a `Result`, because there is no further information in the error case.

### CODE REVIEW

File: src/endpoints/destinations.rs [hunk 1]
```diff
@@ -13,9 +13,13 @@ impl Destination {
...
     ///
-    pub fn from_index(index: usize) -> Destination {
+    pub fn from_index(index: usize) -> Option<Destination> {
         let endpoint_ref = unsafe { MIDIGetDestination(index as ItemCount) };
-        Destination { endpoint: Endpoint { object: Object(endpoint_ref) } }
+        if endpoint_ref == 0 {
+            None
+        } else {
+            Some(Destination { endpoint: Endpoint { object: Object(endpoint_ref) } })
+        }
```
Comment (chris-zen) [inline src/endpoints/destinations.rs]: Would you mind using pattern matching ?

```rust
match endpoint_ref {
  0 => None,
  _ => Some(Destination { endpoint: Endpoint { object: Object(endpoint_ref) } })
}
```

File: src/endpoints/sources.rs [hunk 2]
```diff
@@ -20,9 +20,13 @@ impl Source {
...
     ///
-    pub fn from_index(index: usize) -> Source {
+    pub fn from_index(index: usize) -> Option<Source> {
         let endpoint_ref = unsafe { MIDIGetSource(index as ItemCount) };
-        Source { endpoint: Endpoint { object: Object(endpoint_ref) } }
+        if endpoint_ref == 0 {
+            None
+        } else {
+            Some(Source { endpoint: Endpoint { object: Object(endpoint_ref) } })
+        }
```
Comment (chris-zen) [inline src/endpoints/sources.rs]: Same here.

**Review by chris-zen — CHANGES_REQUESTED** [review]
@Boddlnagg Good catch !
I requested a couple of small changes to use pattern matching, just a personal preference to make the code more readable.
Thank you very much for the contribution !

**Comment by chris-zen** [comment]
@Boddlnagg By the way I just saw that there are tests failing because `from_index` changed to an Option.

**Comment by Boddlnagg** [comment]
I pushed a new version which uses `match` and passes `cargo test`.

**Review by chris-zen — APPROVED** [review]
Great, thanks.

**Pull request merged**